### PR TITLE
feat: fetch accepted events and style upcoming events page

### PR DESCRIPTION
**Pull request by aidenm1**
[//]: # "Feel free to customize this template to your project's vibes"

### :croissant: Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
I created the EventCard component and styled the main webpage with the event cards for the current events in the database.
Updated styling to new version on Figma and changed all styling to use global styling.

### :baguette_bread: Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

image

## :sandwich: How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

First review components/MyEventCard.tsx, then app/events/page.tsx, then app/events/page.style.ts

## :pie: Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Potentially styling? I'm not sure if I did the styling correct

### :pretzel: Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### :bagel: Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @celinechoiii

### CODE REVIEW

File: app/events/page.style.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  layout: responsive;
+  width: 20%;
+  height: 90%;
+`;
+
+export const Page = styled.main`
+  background: ${COLORS.gray1};
+`;
+
+export const Title = styled.main`
```
Comment (celinechoiii) [inline app/events/page.style.ts]: please use the heading available in `styles/text.ts`! to translate `rem` to `px`, usually you just have to multiply it by 16. in this case, since the desired font-size is `32px`, we want heading with font-size `2rem`, which corresponds to `H3`

or, for your component to inherit styles from the `<H3>` tag, you can do the following:

```typescript
import { H3 } from '../../styles/global'; // Import global components

// Extend H3 with additional styles
export const Title = styled(H3)`
  font-style: normal;
  line-height: normal;
  height: 50px;
`;
``` 

For additional context, the `<H3>` tag already has the Bespoke font set as its font-family, and it can accept props for `$fontWeight`, `$color`, and `$align`, so you can further adjust these as needed when using the Title component.

File: app/events/page.style.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+'use client';
```
Comment (celinechoiii) [inline app/events/page.style.ts:1]: rename the file to `style.ts` for consistency

File: app/events/page.tsx [hunk 3]
```diff
@@ -14,37 +16,11 @@ export default function Page() {
...
-              <td>{d.needs_host}</td>
-              <td>{d.performer_type}</td>
-              <td>{d.event_status}</td>
-            </tr>
-          ))}
-        </tbody>
-      </table>
-    </div>
+    <styles.Page>
+      <styles.Title>Upcoming Events</styles.Title>
```
Comment (celinechoiii) [inline app/events/page.tsx]: Replace Title component to `<styles.Title $fontWeight="500" $color="#000" $align="left">Upcoming Events</styles.Title>`

File: components/eventcardcomponent/eventcard.style.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
+'use client';
```
Comment (celinechoiii) [inline components/eventcardcomponent/eventcard.style.ts]: rename the folder to `MyEventCard` and rename the styling file to `style.ts`

**Review by celinechoiii — CHANGES_REQUESTED** [review]

**Comment by celinechoiii** [comment]
overall, amazing job!! i apologize for not bringing up the global styles earlier, but could you update the `styles.ts` file in the event card component folder to align with the changes i noted in the styles file for the event page? let me know if you need further clarifications!

File: app/events/page.tsx [hunk 5]
```diff
@@ -2,9 +2,11 @@
...
 import { fetchAllEvents } from '../../api/supabase/queries/events';
+import MyEventCard from '../../components/eventcardcomponent/MyEventCard';
 import { Event } from '../../types/schema';
+import * as styles from './page.style';
 
-export default function Page() {
+export default function EventPage() {
   const [data, setData] = useState<Event[]>([]);
 
   useEffect(() => {
```
Comment (celinechoiii) [inline app/events/page.tsx:17]: instead of using the `fetchAllEvents` function, write a new SQL query to fetch events by `volunteer_id` and `is_accepted = true` in the `event_status` supabase table

**Comment by celinechoiii** [comment]
LGTM! thank you :))
